src/elexmodel/handlers/data/CombinedData.py

```diff
@@ -74,29 +86,41 @@ def get_reporting_units(self, percent_reporting_threshold, turnout_factor_lower,
         reporting_units["reporting"] = int(1)
         reporting_units["expected"] = True
 
-        return reporting_units
-
-    def get_nonreporting_units(self, percent_reporting_threshold, turnout_factor_lower, turnout_factor_upper):
-        """
-        Get nonreporting data. These are units where expected vote is less than the percent reporting threshold
-        """
         # units where expected vote is less than the percent reporting threshold
         nonreporting_units = self.data[self.data.percent_expected_vote < percent_reporting_threshold].reset_index(
             drop=True
         )
 
-        # remove unexpected units
-        unexpected_units = self._get_unexpected_units(
-            percent_reporting_threshold, turnout_factor_lower, turnout_factor_upper
-        )
+        # remove unexpected and non-predictive units
         nonreporting_units = nonreporting_units[
             ~nonreporting_units.geographic_unit_fips.isin(unexpected_units.geographic_unit_fips)
         ].reset_index(drop=True)
+        nonreporting_units = nonreporting_units[
+            ~nonreporting_units.geographic_unit_fips.isin(non_predictive_units.geographic_unit_fips)
+        ].reset_index(drop=True)
 
         nonreporting_units["reporting"] = int(0)
         nonreporting_units["expected"] = True
 
-        return nonreporting_units
+        # finalize all unexpected/non-predictive units
+        unexpected_units["predictive"] = True
+        non_predictive_units["predictive"] = False
+        all_unexpected_units = pd.concat([unexpected_units, non_predictive_units]).reset_index(drop=True)
+        # since we were not expecting them, we have don't have their county or district
+        # from preprocessed data. so we have to add that back in.
+        if "county_fips" in aggregates:
+            all_unexpected_units["county_fips"] = all_unexpected_units["geographic_unit_fips"].apply(
+                self._get_county_fips_from_geographic_unit_fips
+            )
+        if "district" in aggregates:
+            all_unexpected_units["district"] = all_unexpected_units["geographic_unit_fips"].apply(
+                self._get_district_from_geographic_unit_fips
+            )
```

Comment:
unless I am mistaken, this should only be necessary for the the truly unexpected units, right? not for the non-modeled units?

Reply:
I guess we need to confirm that the non-modeled units actually get the preprocessed data attached.

Reply:
Word, I refactored this so that it would only operate on the truly unexpected units and I had a few unit tests fail, saying these aggregates now contain `nan` for a few units 🤔

Reply:
I assume those unit tests are assuming that non predictive units don't have preprocessed data when we pass it into this function? Or maybe I am wrong from the getgo here

Reply:
No, after making the changes to split the determination of unexpected units from non-modeled units, that should be right.  I think I figured out why that one unit test was failing and then was able to move the `aggregates` part to just `_get_unexpected_units()` after all 😄 🎉